hydra-main/src/main/java/com/addthis/hydra/query/MeshQueryMaster.java

```diff
@@ -246,10 +243,11 @@ protected void writeQuery(ChannelHandlerContext ctx, Query query, ChannelPromise
         if (Strings.isNullOrEmpty(tasks)) {
             return Collections.emptySet();
         } else {
-            return LessStreams.stream(TASKS_SPLITTER.split(tasks))
-                              .map(Ints::tryParse)
-                              .filter(i -> i != null)
-                              .collect(Collectors.toSet());
+            try(Stream<String> stream = LessStreams.stream(TASKS_SPLITTER.split(tasks))) {
+                return stream.map(Ints::tryParse)
+                      .filter(i -> i != null)
+                      .collect(Collectors.toSet());
+            }
```

Comment:
Unlike `Files.list`, most streams have a no-op close, and the API designers actually did not want obsessive resource protection where it was clearly unneeded. Consider, for example, that `stream.map`, and `.filter` also return `Stream` objects that are resources that could be closed.

Personally, I'm not a fan of the arbitrary "does close matter or not -- read the specifics each time and find out!" kind of API, but unless we want to wrap `map` and `filter` and so on, the line has to be drawn somewhere. A private method with a self-contained stream, backed by a simple `String` is one of the most provably harmless cases there is. I can't see the logic in guarding `.stream` in particular, or any of them for that matter.